tests/test_json_schema.py

```diff
@@ -6554,3 +6554,20 @@ class Model(BaseModel):
         some_field: str = Field(alias='_some_field')
 
     assert Model.model_json_schema()['properties']['_some_field']['title'] == 'Some Field'
+
+
+@pytest.mark.skip_json_schema_validation(reason='Custom ref used.')
+def test_arbitrary_ref_in_json_schema() -> None:
+    """See https://github.com/pydantic/pydantic/issues/9981."""
+
+    class Test(BaseModel):
+        x: dict = Field(examples={'$ref': '#/components/schemas/Pet'})
```

Comment:
```suggestion
        x: dict = Field(examples=[{'$ref': '#/components/schemas/Pet'}])
```

Even though the test works because `skip_json_schema_validation` is used, still good to have valid JSON Schemas generated.

Reply:
Or maybe using a dict as an example was a requirement to have the added logic in `get_json_ref_counts` covered by this test?

Reply:
Good call - I've switched to a list, and removed the xfail. This still works, as we unpack a `list` of things in `get_json_ref_counts` and iterate over those :)